dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java

```diff
@@ -835,4 +835,10 @@ public String getDisplayName( String userUid )
     {
         return userDisplayNameCache.get( userUid, c -> userStore.getDisplayName( userUid ) );
     }
+
+    @Override
+    public List<UserCredentials> getUsersWithAuthority( String authority )
+    {
+        return userCredentialsStore.getHasAuthority( authority, Integer.MAX_VALUE );
```

Comment:
I was wondering about the `maxResults` parameter without also having an offset so one could page effectively. If this is used with `Integer.MAX_VALUE` it feels unnecessary to have it. But I might miss another usage.

Reply:
This is only intended as a hard limit to avoid fetching too many recipients when I was doing a search for the messages to check for existence. Since we now use an indexed column for this search, the potential computation has much less impact.
I suggest I just remove this if it is confusing rather than implementing full paging support since that feels pretty out of scope for this PR.